1-Draft/RFC####-Improve-generation-of-argument-string-for-executables.md

```diff
@@ -0,0 +1,85 @@
+---
+RFC: 
+Author: Timo Schwarte
+Status: Draft
+Version: 1.0
+Area: Calling external executables
+Comments Due: 2017-07-07
+Plan to implement: Yes 
+---
+
+# Improve generation of argument-string for executables
+
+When PowerShell calls an external executable, the
+[NativeCommandParameterBinder](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/NativeCommandParameterBinder.cs)
+takes all parameters of the command and generates the argument-string (also called "CommandLine"), which is sent to the executable (as [ProcessStartInfo.Arguments](https://msdn.microsoft.com/en-us/library/system.diagnostics.processstartinfo.arguments.aspx)).
+It seems like the NativeCommandParameterBinder tries to generate an argument-string from the given parameters, such that argv[] of the called executable is set accordingly. This works well, if an argument contains spaces. In this case the NativeCommandParameterBinder adds quotes around the argument, so it appears in `argv[]` as one argument (and is not split into parts).
+However, if the argument itself contains quotes, these are not escaped. Therefore, the corresponding element in `argv[]` has no quotes and depending on the actual string, the argument might be split into multiple `argv[]` elements. It can even occur, that the following arguments are not handled correctly.
+
+This RFC suggests making the NativeCommandParameterBinder compatible to
+[the typical CommandLine escaping rules](https://msdn.microsoft.com/en-us/library/17w5ykft.aspx).
```

Comment:
First things first: Great RFC.

The linked topic's most recent version is now available at https://docs.microsoft.com/en-us/cpp/cpp/parsing-cpp-command-line-arguments

Reply:
Updated.